Automattic/woocommerce-payments
# Update webpack to v5

Now that `@wordpress/dependency-extraction-webpack-plugin` includes the fix I did a while ago is safe to try a update to webpack v5. I opted to do just an update here to make it easier to review and move forward, and do any additional webpack optimization in another PR.

Also want to mention a tricky known behavior of `@wordpress/babel-plugin-makepot`. We need to be careful with webpack cache, because enabling it will break the client pot generation. By default, v5 have in memory cache enabled in dev mode and no cache at all in production.

#### Questions
Have a doubt about the new behavior of v5 by default, that generates a `*.LICENSE.txt` file like the one included below. We can disable this, but I want to ask it here in case we choose to keep them.
```
/*!
 * The buffer module from node.js, for the browser.
 *
 * @author   Feross Aboukhadijeh <feross@feross.org> <http://feross.org>
 * @license  MIT
 */
```

#### Changes proposed in this Pull Request 
- Update webpack to v5 along with used loaders and plugins to ensure compatibility.
- Add needed node polyfills. Webpack 5 doesn't include node polyfills anymore, I tried using `node-polyfill-webpack-plugin` to solve this easily, but there is a known issue with this package that produces errors with the `assert` module.
- Remove unused and deprecated `eslint-loader`, I couldn't find any reference in the project, maybe it's an old dependency?
- Replace `url-loader` with type `asset/inline`, the new way to inline assets in v5.
- Remove unneeded output section. We're using the default configuration here, so no need to declare it.

#### Testing instructions
- Smoke testing basic functionalities to ensure there is no issue after the update. E2E tests already caught an error with the node polyfills 👍 
- Check that `svg` and `png` files are properly inlined.
- Generate a release running `npm run build` before and after applying this branch. They shouldn't be major differences, apart from:
  - All affected `*.php` files will contain a different checksum.
  - All entries JS files `dist/*.js` will be different due to minification.
  - Some `*.css` files will be different by the new assets inlining method.
  - New `*.LICENSE.txt` files will be added.
- Pay special attention to `release/woocommerce-payments/languages/woocommerce-payments.pot`, shouldn't be any difference.
-------------------


**Post merge**
